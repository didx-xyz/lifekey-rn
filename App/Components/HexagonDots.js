// external dependencies
import React from "react"
import { Image } from "react-native"
import PropTypes from "prop-types"

const HexagonDots  = function(props) {
  let image = null

  switch (props.current) {
    case 0:
      image = require("../Images/1.1.gif")
      break
    case 1:
      image = require("../Images/2.1.gif")
      break
    case 2:
      image = require("../Images/3.1.gif")
      break
    case 3:
      image = require("../Images/4.1.gif")
      break
    default:
      image = require("../Images/5.1.gif")
      break
  }

  // switch (props.current) {
  //   case 0:
  //     image = require("../Images/1.gif")
  //     break
  //   case 1:
  //     image = require("../Images/2.gif")
  //     break
  //   case 2:
  //     image = require("../Images/3.gif")
  //     break
  //   case 3:
  //     image = require("../Images/4.gif")
  //     break
  //   default:
  //     image = require("../Images/5.gif")
  //     break
  // }

  return <Image style={{ height: props.height || 67, width: props.width || 67, resizeMode: "contain" }} source={image} />
}

HexagonDots.propTypes = {
  "current": PropTypes.oneOf([0, 1, 2, 3, 4, 5]).isRequired
}

export default HexagonDots
