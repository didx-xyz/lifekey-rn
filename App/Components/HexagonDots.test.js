// external dependencies
import React from "react"

// internal dependencies
import HexagonDots from "./HexagonDots"

test("HexagonDots renders correctly, by default", () => {
  const tree = shallow(
    <HexagonDots />
  )

  expect(tree).toMatchSnapshot()
})

test("HexagonDots renders correctly, when modified", () => {
  const tree = shallow(
    <HexagonDots current={2} />
  )

  expect(tree).toMatchSnapshot()
})
