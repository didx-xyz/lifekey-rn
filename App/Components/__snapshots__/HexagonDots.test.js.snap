// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HexagonDots renders correctly, by default 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;

exports[`HexagonDots renders correctly, when modified 1`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;

exports[`HexagonDots renders correctly, when modified 2`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;

exports[`HexagonDots renders correctly, when modified 3`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;

exports[`HexagonDots renders correctly, when modified 4`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;

exports[`HexagonDots renders correctly, when modified 5`] = `
<Image
  source={1}
  style={
    Object {
      "height": 67,
      "resizeMode": "contain",
    }
  }
/>
`;
